ansible_navigator/configuration/application_configuration.py

```diff
@@ -0,0 +1,199 @@
+""" the ansible-navigator configuration
+"""
+import os
+
+from .application_post_processor import ApplicationPostProcessor
+from .definitions import Config
+from .definitions import CliParameters
+from .definitions import SubCommand
+from .definitions import Entry
+from .definitions import EntryValue
+
+from ..utils import oxfordcomma
+
+
+def abs_user_path(fpath):
+    """Resolve a path"""
+    return os.path.abspath(os.path.expanduser(fpath))
+
+
+def generate_editor_command():
+    """Generate a default for editor_command if EDITOR is set"""
+    if "EDITOR" in os.environ:
+        command = "%s {filename}" % os.environ.get("EDITOR")
+    else:
+        command = "vi +{line_number} {filename}"
+    return command
+
+
+PLUGIN_TYPES = (
+    "become",
+    "cache",
+    "callback",
+    "cliconf",
+    "connection",
+    "httpapi",
+    "inventory",
+    "lookup",
+    "module",
+    "netconf",
+    "shell",
+    "strategy",
+    "vars",
+)
+
+ApplicationConfiguration = Config(
+    application_name="ansible-navigator",
+    post_processor=ApplicationPostProcessor(),
+    subcommands=[
+        SubCommand(name="collections", description="Explore available collections"),
+        SubCommand(name="config", description="Explore the current ansible configuration"),
+        SubCommand(name="doc", description="Review documentation for a module or plugin"),
+        SubCommand(name="inventory", description="Explore an inventory"),
+        SubCommand(name="load", description="Explore a playbook artifact"),
+        SubCommand(name="run", description="Run a playbook"),
+    ],
+    entries=[
+        Entry(
+            name="app",
+            description="Placeholder for subcommand name",
+            subcommand_value=True,
+            value=EntryValue(default="welcome"),
+        ),
+        Entry(
+            name="cmdline",
+            description="Placeholder for argparse remainder",
+            value=EntryValue(default=[]),
+        ),
+        Entry(
+            name="container_engine",
+            choices=["podman", "docker"],
+            cli_parameters=CliParameters(short="--ce"),
+            description="Specify the container engine",
+            value=EntryValue(default="podman"),
+        ),
+        Entry(
+            name="editor_command",
+            cli_parameters=CliParameters(short="--ecmd"),
+            description="Specify the editor comamnd",
+            settings_file_path_override="editor.command",
+            value=EntryValue(default=generate_editor_command()),
```

Comment:
To avoid callable in default this option can be changed to `editor_name` and define a fallback value to read from environment variable

Reply:
by having this here, it will show up when a user does a `--help` if it gets moved to post processing, the user won't see that their EDITOR is the default.